cmd/soroban-rpc/internal/events/events.go

```diff
@@ -261,14 +217,16 @@ func (m *MemoryStore) append(sequence uint32, events []event) error {
 
 	if length < uint32(cap(m.buckets)) {
 		m.buckets = append(m.buckets, bucket{
-			ledgerSeq: sequence,
-			events:    events,
+			ledgerCloseTimestamp: ledgerCloseTimestamp,
```

Comment:
nice to have: move the bucket creation ahead of the if statement, so that you won't need to repeat the bucket creation on either of the cases.